02-microarray/differential-expression_microarray_01.Rmd

```diff
@@ -0,0 +1,349 @@
+
+```{r}
+# Read in metadata TSV file
+metadata <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your metadata file
+  "metadata_GSE71270.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data TSV file
+df <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your data file
+  "GSE71270.tsv" # Replace with the name of your data file
+)) %>%
+  # Tuck away the Gene id column as rownames
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>%
+  dplyr::select(metadata$geo_accession)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$geo_accession)
+```
+
+## Set up design matrix
+
+`limma` needs a numeric design matrix to signify which are CREB and control samples. 
+Here we are using the treatments supplied in the metadata to create a design matrix where the "none" samples are assigned `0` and the "amputated" samples are assigned `1`.
+Note that the metadata variables that signify the treatment groups might be different in across datasets and might not always be underneath the  category.
+
+The `genotype/variation` column contains group information we will be using for differential expression. 
+But the `/` it contains in its column name makes it more annoying to access.  
+Accessing variable that have names with special characters like `/`, or spaces, require extra work-arounds to ignore R's normal interpretations of these characters.
+
+```{r}
+metadata <- metadata %>%
+  dplyr::rename("genotype" = `genotype/variation`) # This step will not be the same (or might not be needed at all) with a different dataset
+```
+
+Now we will create a model matrix based on our newly renamed `genotype` variable. 
+
+```{r}
+# Create the design matrix from the genotype information
+des_mat <- model.matrix(~ metadata$genotype)
+```
+
+## Apply linear model
+
+After applying our data to linear model, in this example we apply empirical Bayes smoothing and Benjamini-Hochberg multiple testing correction. 
+The `topTable()` function default is to use Benjamini-Hochberg but this can be changed to a different method using the `adjust.method` argument.
+
+```{r}
+# Apply linear model to data
+fit <- lmFit(df, design = des_mat)
+
+# Apply empirical Bayes to smooth standard errors
+fit <- eBayes(fit)
+
+# Apply multiple testing correction and obtain stats
+stats_df <- topTable(fit, number = nrow(df)) %>%
+  tibble::rownames_to_column("Gene")
+```
+
+Let's take a peek at what our results table looks like. 
+
+```{r}
+stats_df
+```
+
+By default, results are ordered by largest `B` (the log odds value) to the smallest, which means your most differentially expressed genes should be toward the top. 
+
+See the help page by using `?topTable` for more information and options for this table. 
```

Comment:
```suggestion
See the help page by using `?topTable` for more information on arguments like `adjust.method` and options for this table.
```

This is not completely necessary but I thought it would be helpful to explicitly include a note that more info on the `adjust.method` argument that is mentioned on line 240 (but not actually used here) can also be found by using `topTable`.

Reply:
Hmm... I don't feel like we need to rehash about `adjust.method` as an argument. This is enough to get them to the help page and they can find out about that from there. But I will add a reminder about the help page after I tell them about `adjust.method` the first time.